lib/ex_crypto.ex

```diff
@@ -36,6 +36,33 @@ defmodule ExCrypto do
     {kind, Exception.normalize(kind, error), System.stacktrace}
   end
 
+  @aes_block_size 16
+  @iv_bit_length 128
+
+  defp key_and_iv_for_algorithm(raw_key, raw_iv, algorithm) do
```

Comment:
I'm not sure we can assume base64 for incoming data.  There is a helper functions here for payloads called `encode_payload` and `decode_payload` for use as helpers for this sort of thing. Otherwise we'll probably need to just treat everything as binaries because it's just as likely that something has been base16 encoded as it is that it's been base64 encoded.